### PR TITLE
Adds std::duration::Duration from/to Python conversions  

### PR DESCRIPTION
**Pull request by Tpt**

### CODE REVIEW

**Comment by Tpt** [comment]
We already support `chrono::Duration`, it might be nice to support `std::duration::Duration` for consistency.

Things to notice:
- `std::duration::Duration` only supports positive durations, an exception is raised when converting a negative `timedelta`
- a warning is created if precision is lost when converting from `Duration` to `timedelta`, opposite to `chrono::Duration` (is it a good idea?)

File: src/tests/common.rs [hunk 1]
```diff
@@ -129,9 +129,9 @@ mod inner {
     #[macro_export]
     macro_rules! assert_warnings {
         ($py:expr, $body:expr, [$(($category:ty, $message:literal)),+] $(,)? ) => {{
-            CatchWarnings::enter($py, |w| {
+            $crate::tests::common::CatchWarnings::enter($py, |w| {
                 $body;
-                let expected_warnings = [$((<$category>::type_object($py), $message)),+];
+                let expected_warnings = [$((<$category as $crate::type_object::PyTypeInfo>::type_object($py), $message)),+];
```
Comment (Tpt) [inline src/tests/common.rs:134]: I allowed myself to make this macro more portable

**Review by adamreichold — APPROVED** [review]
LGTM

**Review by davidhewitt — APPROVED** [review]
👍 likewise, thanks!

It's a little sad that Rust stdlib duration is strictly positive, but that's not really our design problem to worry about here.

**Comment by davidhewitt** [comment]
Hmm, there are some test failures?

> * a warning is created if precision is lost when converting from `Duration` to `timedelta`, opposite to `chrono::Duration`. Is it a good idea?

Yeah I think that's fine to warn 👍

**Comment by adamreichold** [comment]
@Tpt Argh, unused import on `wasm32-wasi` derailed the CI...

**Comment by adamreichold** [comment]
Pushed a fixup.

**Pull request merged**